jackson/src/main/java/com/baeldung/jackson/deserialization/enums/customdeserializer/CustomEnumDeserializer.java

```diff
@@ -0,0 +1,42 @@
+package com.baeldung.jackson.deserialization.enums.customdeserializer;
+
+import java.io.IOException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+public class CustomEnumDeserializer extends StdDeserializer<Distance> {
+
+    private static final long serialVersionUID = -1166032307856492833L;
+
+    public CustomEnumDeserializer() {
+        this(null);
+    }
+
+    public CustomEnumDeserializer(Class<?> vc) {
+        super(vc);
+    }
+
+    @Override
+    public Distance deserialize(final JsonParser jsonParser, final DeserializationContext ctxt) throws IOException, JsonProcessingException {
```

Comment:
If the final keyword isn't required, then let's omit it as per the coding guidelines.